src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -394,37 +377,37 @@ private void DoCheckTimeout()
         /// then to leave runtextpos at the ending position. It should leave
         /// runtextpos where it started if there was no match.
         /// </summary>
-        protected abstract void Go();
+        protected virtual void Go() => throw new NotImplementedException();
 
         /// <summary>
         /// The responsibility of FindFirstChar() is to advance runtextpos
         /// until it is at the next position which is a candidate for the
         /// beginning of a successful match.
         /// </summary>
-        protected abstract bool FindFirstChar();
+        protected virtual bool FindFirstChar() => throw new NotImplementedException();
 
         /// <summary>
         /// InitTrackCount must initialize the runtrackcount field; this is
         /// used to know how large the initial runtrack and runstack arrays
         /// must be.
         /// </summary>
-        protected abstract void InitTrackCount();
+        protected virtual void InitTrackCount() => runtrackcount = 0;
```

Comment:
Does this need a body or can it be empty?

Reply:
It could be empty. I just set it to zero as that was in your original proposal.